### ISSUE TITLE
Error: 1002: Verification Error: Execution: Could not convert parameter tx between node and runtime: No such variant in enum MultiAddress

### ISSUE THREAD

**Issue by tansaku** [paritytech/canvas-ui#67]
Hi folks, so I'm a bit stuck on the smart ink tutorial - I get an error trying to upload my flipper contract as described in 

https://substrate.dev/substrate-contracts-workshop/#/0/deploying-your-contract

I get:

> Transaction Failed: Error: 1002: Verification Error: Execution: Could not convert parameter tx between node and runtime: No such variant in enum MultiAddress: RuntimeApi, Execution: Could not convert parameter tx between node and runtime: No such variant in enum MultiAddress

which interestingly is the same error that I got when my local frontend was 2.0.0 and my backend was 2.0.1 as described here:

https://github.com/substrate-developer-hub/substrate-front-end-template/issues/158

More details on the issue in:

https://github.com/substrate-developer-hub/substrate-contracts-workshop/issues/100

is there some way to upgrade https://paritytech.github.io/canvas-ui to work with 2.0.1 backends, or maybe I can run https://paritytech.github.io/canvas-ui locally and upgrade myself?

**Comment by tansaku** [comment]
thanks @kwingram25 - I've had a first stab, but failed in https://github.com/paritytech/canvas-ui/pull/68 - what should I be reading to try and understand where this multiaddress setting should be added to canvas-ui?

**Comment by kwingram25** [comment]
Normally this kind of problem means there's an update to polkadot-js/api that needs to be added here. I will take a look myself but you are on the right track - guessing it is a change to MultiAddress type that needs to be accounted for

**Comment by tansaku** [comment]
thanks @kwingram25 - if you have a moment can you point me at any docs that might help me better understand the relation between polkadot-js/apps, canvasui and the substrate-front-end-template - they are all front ends to connect with a back end polkadot network right?

If I wanted to better understand the structure or canvas-ui is the best approach to just start reading the sourcecode? or I should start with polkadot-js/apps or refer to a wiki or other docs somewhere else?

**Comment by kwingram25** [comment]
Hi! I've been trying to reproduce the issue but am having trouble. Are you trying to use Canvas UI with Substrate? I am seeing issues in that regard but not the same ones - it continues to work for me connecting to Canvas Node.

I would recommend looking at the polkadot-js/apps source code and particularly react-api package as that is where the chain types come into play.

**Comment by tansaku** [comment]
thanks hi @kwingram25 thanks for coming back to me.  I am trying to use:

> Canvas UI with Substrate

I am not sure what "Canvas Node" is.  That's another back end?

I followed the substrate tutorials to set up the substrate-node-template, for which the 2.0.0 wouldn't run on my OSX, so I had to bump to 2.0.1 and then that required a fix on the front end as mentioned in https://github.com/substrate-developer-hub/substrate-front-end-template/issues/158

Thanks for the tips as to where to look in the source code.

It sounds like the Canvas UI is designed to run with the Canvas Node, and not with the substrate node template?   Are substrate and canvas two fundamentally different things?
